express-api/src/services/projects/projectsServices.ts

```diff
@@ -123,8 +205,188 @@ const addProjectBuildingRelations = async (project: Project, buildingIds: number
   );
 };
 
+/**
+ * Removes the relations between a project and the specified parcel IDs.
+ *
+ * @param {Project} project - The project from which to remove the parcel relations.
+ * @param {number[]} parcelIds - An array of parcel IDs to remove the relations for.
+ * @returns {Promise<void>} - A promise that resolves when the relations have been removed.
+ */
+const removeProjectParcelRelations = async (project: Project, parcelIds: number[]) => {
+  await Promise.all(
+    parcelIds?.map(async (parcelId) => {
+      await projectPropertiesRepo.delete({
+        ProjectId: project.Id,
+        ParcelId: parcelId,
+      });
+    }),
+  );
+};
+
+/**
+ * Removes the relationship between a project and the specified buildings.
+ *
+ * @param {Project} project - The project from which to remove the building relationships.
+ * @param {number[]} buildingIds - An array of building IDs to be removed from the project.
+ * @returns {Promise<void>} - A promise that resolves when the building relationships have been removed.
+ */
+const removeProjectBuildingRelations = async (project: Project, buildingIds: number[]) => {
+  await Promise.all(
+    buildingIds?.map(async (buildingId) => {
+      await projectPropertiesRepo.delete({
+        ProjectId: project.Id,
+        BuildingId: buildingId,
+      });
+    }),
+  );
+};
+
+/**
+ * Updates a project with the given changes and property IDs.
+ *
+ * @param project - The project object containing the changes to be made.
+ * @param propertyIds - The IDs of the properties to be associated with the project.
+ * @returns The result of the project update.
+ * @throws {ErrorWithCode} If the project name is empty or null, if the project does not exist, if the project number or agency cannot be changed, or if there is an error updating the project.
+ */
+const updateProject = async (project: DeepPartial<Project>, propertyIds: ProjectPropertyIds) => {
```

Comment:
Hit the following error 
```
{"level":"info","message":"query: START TRANSACTION","timestamp":"2024-04-17 01:49:26.892 PM"}
{"level":"info","message":"query: ROLLBACK","timestamp":"2024-04-17 01:49:26.892 PM"}
{"level":"warn","message":"Cannot query across one-to-many for property ProjectProperties","timestamp":"2024-04-17 01:49:26.893 PM"}
{"level":"error","message":"Error updating project.","timestamp":"2024-04-17 01:49:26.894 PM"}
```

when trying to run this query:
```
const sample_project = await projectServices.getProjectById(9);
const result = await projectServices.updateProject(sample_project, {parcels:[7479], buildings:[]});
```

Reply:
I think this one is now solved...
Had to change the .update to .save. It seems to expect something different in terms of what updated properties you  can pass.

Reply:
no longer hitting this error!